src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,64 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final String COMMA = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] infoFromFileName = getInfoFromFileName(fromFileName);
+        String reportSupplyAndBuy = getInfoSupplyAndBuy(infoFromFileName);
+        writeToFile(toFileName, reportSupplyAndBuy);
+    }
+
+    public String[] getInfoFromFileName(String fromFileName) {
+        File file = new File(fromFileName);
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+            String value = reader.readLine();
+            while (value != null) {
+                stringBuilder.append(value).append(System.lineSeparator());
+                value = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file - " + fromFileName, e);
+        }
+        String sentence = stringBuilder.toString();
+        return sentence.split(System.lineSeparator());
+    }
+
+    public String getInfoSupplyAndBuy(String[] split) {
```

Comment:
let's make method name more concise and clear, also pay attention to access modifiers and parameter name